scalafix-tests/unit/src/test/scala/scalafix/tests/cli/CliGitDiff.scala

```diff
@@ -0,0 +1,173 @@
+package scalafix.tests.cli
+
+import java.lang.ProcessBuilder
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.PrintStream
+import java.nio.charset.StandardCharsets
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Paths, Path, StandardOpenOption}
+
+import org.scalactic.source.Position
+import org.scalatest._
+import org.scalatest.FunSuite
+
+import scala.collection.immutable.Seq
+import scala.util._
+import scalafix.cli
+import scalafix.internal.cli.CommonOptions
+import scalafix.testkit.DiffAssertions
+
+class CliGitDiff() extends FunSuite with DiffAssertions {
+  gitTest("it should handle addition") { (fs, git, cli) =>
+    val oldCode = "old.scala"
+    val newCode = "new.scala"
+    val newCodeAbsPath = fs.absPath(newCode)
+
+    val confFile = ".scalafix.conf"
+
+    git.init()
+    fs.add(
+      oldCode,
+      """|object OldCode {
+         |  // This is old code, where var's blossom
+         |  var oldVar = 1
+         |}""".stripMargin)
+    git.add(oldCode)
+    fs.add(
+      confFile,
+      """|rules = DisableSyntax
+         |DisableSyntax.keywords = [var]""".stripMargin)
+    git.add(confFile)
+    git.commit()
+
+    git.checkout("pr-1")
+
+    fs.add(
+      newCode,
+      """|object NewCode {
+         |  // New code, no vars
+         |  var newVar = 1
+         |}""".stripMargin)
+    git.add(newCode)
+    git.commit()
+
+    val obtained = noColor(cli.run("--diff"))
+
+    val expected =
+      s"""|$newCodeAbsPath:3: error: [DisableSyntax.keywords.var] keywords.var is disabled
```

Comment:
😍 👏 Nice!

One trick to avoid the `$newCodeAbsPath` splice is to `replace(tmpDirectory.toString, "<...>")`. I find it makes the test readable, but this is fine too